### PR TITLE
Fix bug on clear

### PR DESCRIPTION
**Pull request by huanghao1998**
What's change
- `Clear` now only clears contact list #237 
   -  Add removeAllMemberInproject(): using removeIf to remove member from the given predicate
- `Clear t/` now about to clear respective members #238 
   -  Instead of setSocket to a `new Socket()`, it will remove all the person in the list

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -73,7 +73,8 @@ You can use the links in the Table of Contents to quickly navigate through this
   e.g. in `add n/NAME`, `NAME` is a parameter which can be used as `add n/John Doe`.
 
 * Keywords in `UPPER_CASE` supplied to remove commands `remove` and `removepj`, are optional.<br>
-  e.g. in `remove 1 p/PHONE`, `PHONE` can be left empty.
+  e.g. in `remove 1 p/PHONE`, `PHONE` can be left empty.<br>
+  **Note:** If keyword is left empty, SOCket will assume that it has removed an empty field so error message will not be shown even if there is a wrong input of keyword in the subsequent prefix.
```
Comment (chia-yh) [inline docs/UserGuide.md:77]: "so an error message"

"there is a wrong keyword"

"in a subsequent prefix"

File: docs/UserGuide.md [hunk 2]
```diff
@@ -177,6 +178,8 @@ Examples:
   </p>
 
 
+
+
```
Comment (chia-yh) [inline docs/UserGuide.md:182]: what are these for

File: docs/UserGuide.md [hunk 3]
```diff
@@ -164,7 +165,7 @@ Format: `remove INDEX [p/[PHONE]] [p/[GITHUBPROFILE]] [e/[EMAIL]] [a/[ADDRESS]]
 * At least one of the optional fields must be provided.
 * Removes the corresponding field value(s) matching the `KEYWORD`.
   * e.g. `l/Java` will remove `Java` from the languages of the contact.
-* Clears the field value(s) when no keyword is provided for the field.
+* Clears the field value(s) when no keyword is provided for the field even if the field is empty.
```
Comment (chia-yh) [inline docs/UserGuide.md:168]: Perhaps consider extracting the added section as a tip or note instead; e.g. "`remove` will remove a field successfully even if it is empty" or something

File: docs/images/AssignAfter.png
Comment (chia-yh) [inline docs/images/AssignAfter.png:1]: For the UI images, are there any plans to focus on the particular portions of the UI that change as discussed? For example, AssignBefore and AssignAfter could focus on the project card that the person is assigned to, and could include the expected output in text somewhere; similarly for other commands

**Review by chia-yh — CHANGES_REQUESTED** [review]

File: docs/images/AssignAfter.png
Comment (chia-yh) [inline docs/images/AssignAfter.png:1]: I like that you added the callouts, but I thought what was suggested was to crop to a smaller portion of the UI, so formatting around the pictures would be easier. Could I get your thoughts on this?

File: docs/UserGuide.md [hunk 4]
```diff
@@ -167,6 +168,10 @@ Format: `remove INDEX [p/[PHONE]] [p/[GITHUBPROFILE]] [e/[EMAIL]] [a/[ADDRESS]]
 * Clears the field value(s) when no keyword is provided for the field.
   * e.g. `t/` will remove all the tags associated with the contact.
 
+<div markdown="span" class="alert alert-primary"> :bulb: Tip
+`remove` command considers **removing a field successfully** when no keyword is provided even if the field is empty.
```
Comment (chia-yh) [inline docs/UserGuide.md:172]: Perhaps consider "If no keyword is provided, `remove` will still execute successfully on an empty field"

Just some concerns as describing this behavior is rather hard

**Review by chia-yh — CHANGES_REQUESTED** [review]
Some changes to suggest, especially in the case of UI screenshots in relation to formatting problems

**Comment by chia-yh** [comment]
LGTM, however, if you're adamant on the full UI screenshots, you're going to have to find an alternative unless we have the formatting be as bad as it is now. Personally, I don't see an issue with cropping to the relevant areas of the UI, and as mentioned, including either the output as text or a separate cropped image; either way would be easier to manage

**Comment by huanghao1998** [comment]
- fixes #237 
- fixes #238

**Pull request merged**